src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp

```diff
@@ -539,14 +554,19 @@ real64 CompositionalMultiphaseBase::solverStep( real64 const & time_n,
 {
   GEOSX_MARK_FUNCTION;
 
-  real64 dt_return;
-
-  setupSystem( domain, m_dofManager, m_localMatrix, m_localRhs, m_localSolution );
+  // Only build the sparsity pattern once
+  // TODO: this should be triggered by a topology change indicator
+  static bool systemSetupDone = false;
+  if( !systemSetupDone )
+  {
+    setupSystem( domain, m_dofManager, m_localMatrix, m_localRhs, m_localSolution );
+    systemSetupDone = true;
+  }
```

Comment:
To be honest, this was added as a temporary plug, because system setup is too slow compared to the rest of the time step when profiling (i.e. with the linear solve disabled), and results in constant data movement. I decided to keep it, because it's going to be useful during further optimization work (if any), and for the time being we are not doing any topology change (fracture propagation) with multiphase flow. We're definitely gonna need some dynamic topology change flag in `MeshLevel` or elsewhere.